pynest/pynestkernel.pyx

```diff
@@ -623,12 +623,14 @@ cdef inline object sli_datum_to_object(Datum* dat):
 
 cdef inline object sli_array_to_object(ArrayDatum* dat):
 
-    cdef tmp = [None] * dat.size()
+    cdef tmp = [None] * int(dat.size())
```

Comment:
Is `int` big enough or should it rather be cast to `size_t`?

Reply:
Casting it to `int` was the only solution to the problem. This is probably due to a Cython bug related to the `len( )` function that will be solved in [Cython 3.0.0 ](https://cython.readthedocs.io/en/latest/src/changes.html)

A working alternative can be:
```
    cdef Py_ssize_t i
    cdef Py_ssize_t n = dat.size()

    cdef tmp = [None] * n
    cdef Token* tok = dat.begin()


    for i in range(n):
        tmp[i] = sli_datum_to_object(tok.datum())
        inc(tok)
```
Do you prefer this option?

Reply:
I think that's fine. At least I definitely like it better than the idea of the original solution behaving randomly or (IMO actually safer) crashing when converting `ArrayDatum`s of a size larger than what an `int` can hold...

However, it is probably a good idea to also add a short comment explaining why these steps are needed to stop the next someone from "simplifying" it in a year from now ;-)

Reply:
Hi, as far as I'm aware, Python's "int" is not a primitive type, but some clever contraption that has no upper bound. However, I'm not sure whether this line of code will be executed by the Python interpreter, or first converted into C and run there. It might be instructive to look at the generated code.

Reply:
These are the relative generated codes [before ](https://pastebin.com/wjNKzPx9) and [after ](https://pastebin.com/pBG0FePE) the PR.

Reply:
> I think that's fine. At least I definitely like it better than the idea of the original solution behaving randomly or (IMO actually safer) crashing when converting `ArrayDatum`s of a size larger than what an `int` can hold...
> 
> However, it is probably a good idea to also add a short comment explaining why these steps are needed to stop the next someone from "simplifying" it in a year from now ;-)

I have added a couple of lines of comment ;-)

Reply:
OK, it looks like neither version is doing any downcasting to machine word lengths! Readability of the generated code could be improved, but I'll leave this verdict up to @jougs. :)